File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/TabHorizontalLayoutDemoActivity.java
package io.ganguo.demo.view.activity;

import android.content.Context;
import android.content.Intent;

import io.ganguo.demo.R;
import io.ganguo.demo.databinding.ActivityTabLayoutDemoBinding;
import io.ganguo.demo.view.activity.base.GGVModelActivity;
import io.ganguo.demo.viewmodel.activity.TabHorizontalLayoutDemoVModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.vmodel.ViewModelHelper;
import io.ganguo.viewmodel.common.HeaderItemViewModel;
import io.ganguo.viewmodel.common.HeaderViewModel;

/**
 * Created by leo on 16/7/12.
 * TabLayoutDemo - activity
 */
public class TabHorizontalLayoutDemoActivity extends GGVModelActivity<ActivityTabLayoutDemoBinding, TabHorizontalLayoutDemoVModel> {
    public static Intent intentFor(Context context) {
        Intent intent = new Intent(context, TabHorizontalLayoutDemoActivity.class);
        return intent;
    }

    @Override
    public TabHorizontalLayoutDemoVModel createViewModel() {
        return new TabHorizontalLayoutDemoVModel();
    }

    @Override
    public void onViewAttached(TabHorizontalLayoutDemoVModel viewModel) {
        ViewModelHelper.bind(getBinding().includeHeader, viewModel, onCreateHeaderVModel());
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/TabHorizontalLayoutDemoActivity.kt
package io.ganguo.demo.view.activity

import android.content.Context
import android.content.Intent

import io.ganguo.demo.databinding.ActivityTabLayoutDemoBinding
import io.ganguo.demo.view.activity.base.GGVModelActivity
import io.ganguo.demo.viewmodel.activity.TabHorizontalLayoutDemoVModel
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.ViewModelHelper

/**
 * Created by leo on 16/7/12.
 * TabLayoutDemo - activity
 */
class TabHorizontalLayoutDemoActivity : GGVModelActivity<ActivityTabLayoutDemoBinding, TabHorizontalLayoutDemoVModel>() {

    override fun createViewModel(): TabHorizontalLayoutDemoVModel {
        return TabHorizontalLayoutDemoVModel()
    }


    override fun onViewAttached(viewModel: TabHorizontalLayoutDemoVModel) {
        ViewModelHelper.bind(binding.includeHeader, viewModel, newHeaderVModel())
    }


    companion object {
        fun intentFor(context: Context): Intent {
            return Intent(context, TabHorizontalLayoutDemoActivity::class.java)
        }
    }
}
